src/device_batch_gemm.cc

```diff
@@ -82,13 +82,16 @@ void gemm(
         device_blas_int ldb_ = to_device_blas_int( ldb[0] );
         device_blas_int ldc_ = to_device_blas_int( ldc[0] );
 
-        size_t batch_limit = queue.get_batch_limit();
-        scalar_t** dAarray = (scalar_t**) queue.get_dev_ptr_array( );
-        scalar_t** dBarray = dAarray + batch_limit;
-        scalar_t** dCarray = dBarray + batch_limit;
+        // gemm needs 3 arrays (A, B, and C).
+        size_t max_chunk = MaxBatchChunk;
+        queue.work_resize<void*>( 3*max_chunk );
```

Comment:
Should not we resize only when we do not have enough space?

Reply:
Per discussion, changed to `work_ensure_size`.